pkg/x/sylt/terra_state.go

```diff
@@ -0,0 +1,80 @@
+package sylt
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/golingon/lingon/pkg/terra"
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+// StateStatus defines how complete the state is for a stack.
+type StateStatus int
+
+const (
+	// StateStatusUnknown the state mode has not been determined yet
+	// (e.g. no plan/apply).
+	StateStatusUnknown StateStatus = 0
+	// StateStatusEmpty there is no state (e.g. no apply yet).
+	StateStatusEmpty StateStatus = 1
+	// StateStatusPartial there is a state, but there are resources in the Stack
+	// that are not in the state yet (e.g. need to be applied).
+	StateStatusPartial StateStatus = 2
+	// StateStatusFull all resources in the Stack have state.
+	StateStatusFull StateStatus = 3
+)
+
+// StackImportState imports the Terraform state into the Terraform Stack.
+// A bool is returned indicating whether all the resources have state. If the
+// bool is true, every resource in the stack will have some state. If the bool
+// is false, the state is incomplete meaning some resources may have state.
+func StackImportState(
+	stack terra.Exporter,
+	state *tfjson.State,
+) (StateStatus, error) {
+	if state.Values == nil || len(state.Values.RootModule.Resources) == 0 {
+		return StateStatusEmpty, nil
+	}
+	sb, err := terra.ObjectsFromStack(stack)
+	if err != nil {
+		return StateStatusUnknown, fmt.Errorf("getting stack objects: %w", err)
+	}
+	isFullState := true
+	stateResources := state.Values.RootModule.Resources
+	// Iterate over the resources in the Stack and try to find the corresponding
+	// resource in the state.
+	// If it exists, import the state into the Stack.
+	for _, res := range sb.Resources {
+		resFound := false
+		for _, sr := range stateResources {
```

Comment:
could it be that some resources in the state are not in the stack ?

Reply:
Yes, it is possible. If a plan&apply has not been performed and we are just checking the state, for example.

Reply:
Added a status for this, `StateStatusOverflow`. Perhaps not the best name but the best I could do right now.